### PR TITLE
[SPARK-30262][SQL] Avoid NumberFormatException when totalSize is empty

### PR DESCRIPTION
**Pull request by southernriver**
### What changes were proposed in this pull request?

We could get the Partitions Statistics Info.But in some specail case, The Info  like  totalSize，rawDataSize，rowCount maybe empty. When we do some ddls like   
`desc formatted partition` ,the NumberFormatException is showed as below:
```
spark-sql> desc formatted table1 partition(year='2019', month='10', day='17', hour='23');
19/10/19 00:02:40 ERROR SparkSQLDriver: Failed in [desc formatted table1 partition(year='2019', month='10', day='17', hour='23')]
java.lang.NumberFormatException: Zero length BigInteger
at java.math.BigInteger.(BigInteger.java:411)
at java.math.BigInteger.(BigInteger.java:597)
at scala.math.BigInt$.apply(BigInt.scala:77)
at org.apache.spark.sql.hive.client.HiveClientImpl$$anonfun$31.apply(HiveClientImpl.scala:1056)
```
Although we can use 'Analyze table partition ' to update the totalSize,rawDataSize or rowCount, it's unresonable for normal SQL to throw NumberFormatException for Empty totalSize.We should fix the empty case when readHiveStats.

### Why are the changes needed?

This is a related to the robustness of the code and may lead to unexpected exception in some unpredictable situation.Here is the case:
image

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

manual

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Thank you for your first contribution, @southernriver .

**Comment by dongjoon-hyun** [comment]
ok to test

File: sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala [hunk 1]
```diff
@@ -1039,9 +1039,10 @@ private[hive] object HiveClientImpl {
    * Note that this statistics could be overridden by Spark's statistics if that's available.
    */
   private def readHiveStats(properties: Map[String, String]): Option[CatalogStatistics] = {
-    val totalSize = properties.get(StatsSetupConst.TOTAL_SIZE).map(BigInt(_))
-    val rawDataSize = properties.get(StatsSetupConst.RAW_DATA_SIZE).map(BigInt(_))
-    val rowCount = properties.get(StatsSetupConst.ROW_COUNT).map(BigInt(_))
+    val totalSize = properties.get(StatsSetupConst.TOTAL_SIZE).filter(_.nonEmpty).map(BigInt(_))
+    val rawDataSize = properties.get(StatsSetupConst.RAW_DATA_SIZE).filter(_.nonEmpty)
+      .map(BigInt(_))
+    val rowCount = properties.get(StatsSetupConst.ROW_COUNT).filter(_.nonEmpty).map(BigInt(_))
```
Comment (dongjoon-hyun) [inline sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: This is a logical patch, but there is no evidence when this happens.
Could you give me a reproducible procedure which you met?

Comment (dongjoon-hyun) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: I'm wondering if you are hitting the other system's bug. Then, you had better fix the root cause.
> This is a related to the robustness of the code and may lead to unexpected exception in some unpredictable situation.Here is the case:

Comment (maropu) [reply sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala]: Also, can you add tests for that?

**Review by srowen — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
Okay. Since this is a trivial safeguard, I'll merge this to master. Thank you, @southernriver , @srowen , @maropu .

**Comment by dongjoon-hyun** [comment]
@southernriver . You are added to the Apache Spark contributor group and SPARK-30262 is assigned to you. Thank you again.

**Pull request closed without merging**